Backend/dependencies.py

```diff
@@ -0,0 +1,32 @@
+from fastapi import Depends
+
+from database.firestore.database_manager import DatabaseManager
+from interfaces.statistic import IStatisticService
+from services.task import TaskService
+from services.statistic import StatisticService
+from interfaces.task import ITaskService
+
+
+async def get_db_manager() -> DatabaseManager:
+    try:
+        db_manager = DatabaseManager()
+        return db_manager
+    except Exception as e:
+        raise RuntimeError(f"Error creating DatabaseManager: {e}")
+
+
+def get_task_service(
+        database_manager: DatabaseManager = Depends(get_db_manager)
+) -> ITaskService:
+    task_service: ITaskService = TaskService(database_manager, "tasks")
+    yield task_service
+
+
+def get_statistic_service(
+        task_service: ITaskService = Depends(get_task_service)
+) -> IStatisticService:
+    statistic_service: IStatisticService = StatisticService(task_service=task_service)
+    yield statistic_service
+
+
+db_manager_dependency = Depends(get_db_manager)
```

Comment:
:question: Why is this `db_manager_dependency` initialized here?

Reply:
for future if we want to handle with another database manager, going with this flow we should delete whole dependencies.py file because we have only one type of database right now